### PR TITLE
Binary Search √

### PR DESCRIPTION
**Pull request by mixelpixel**
mind === blown // <--- true

### CODE REVIEW

**Comment by mixelpixel** [comment]
sorry for all the commits, just doing a little GitHub "latest commit msg" prettyfying 🍔

File: binarySearch/binarySearch.js [hunk 1]
```diff
@@ -10,6 +10,49 @@
...
 
+/* FUNCTION:
+  * We know the target is at one of the array indexes. Take the average of min & max range, divide by half
+    * If target number at that index, return index #
+    * Is the value at that index   less  than the target? Then increase the minimum to the guess index position + 1
+    * Is the value at that index greater than the target? Then decrease the maximum to the guess index position - 1
+*/
+
+const binarySearch = (nums, target) => {
+  if (nums.length < 1) return "Error001: empty array";
```
Comment (ryan-hamblin) [inline binarySearch/binarySearch.js:32]: Error001 sounds like a college class.

File: binarySearch/binarySearch.js [hunk 2]
```diff
@@ -10,6 +10,49 @@
...
+  let guess;
+  while (min <= max) {
+    guess = Math.floor((min + max) / 2);
+    if (nums[guess] === target) { return guess }
+    else if (nums[guess] < target) { min = guess + 1 }
+    else { max = guess - 1 }
+  }
 };
+
+// // TEST SUITE
```
Comment (ryan-hamblin) [inline binarySearch/binarySearch.js:45]: I love that you do this! It's a solid way to demonstrate your code and how it's operating/running Great work!

File: binarySearch/binarySearch.js [hunk 3]
```diff
@@ -10,6 +10,49 @@
...
+    * Is the value at that index greater than the target? Then decrease the maximum to the guess index position - 1
+*/
+
+const binarySearch = (nums, target) => {
+  if (nums.length < 1) return "Error001: empty array";
+  if (nums.indexOf(target) === -1) return "Error002: target number not in array";
+  let min = 0;
+  let max = nums.length - 1;
+  let guess;
+  while (min <= max) {
```
Comment (ryan-hamblin) [inline binarySearch/binarySearch.js:37]: I love a solid iterative approach to these problems! Nicely done

**Pull request closed without merging**